src/main/kotlin/org/rust/lang/core/completion/RustCompletionEngine.kt

```diff
@@ -23,9 +24,12 @@ object RustCompletionEngine {
 
     fun completeFieldOrMethod(field: RustFieldExprElement): Array<out LookupElement> {
         val dispatchType = field.expr.resolvedType.stripAllRefsIfAny()
-        val methods = dispatchType.nonStaticMethods
+
         // Needs type ascription to please Kotlin's type checker, https://youtrack.jetbrains.com/issue/KT-12696.
         val fields: List<RustNamedElement> = (dispatchType as? RustStructType)?.struct?.fields.orEmpty()
+
+        val methods = RustImplIndex.findNonStaticMethodsFor(dispatchType, field.project)
```

Comment:
Finding methods is an API of the `RustType`. The fact that we use an index for this is an implementation detail. And we want to use dynamic dynamic dispatch here to avoid `is RustTypeParameterType` cast.